src/python/Utils/TokenManager.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#pylint: disable=W0212
+"""
+File       : TokenManager.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: IAM token manager
+"""
+
+# system modules
+import os
+import ssl
+import time
+import logging
+import traceback
+
+# third part library
+try:
+    import jwt
+except ImportError:
+    traceback.print_exc()
+    jwt = None
+
+# prevent "SSL: CERTIFICATE_VERIFY_FAILED" error
+# this will cause pylint warning W0212, therefore we ignore it above
+ssl._create_default_https_context = ssl._create_unverified_context
+
+
+def getToken(name=None):
+    """
+    Get IAM token either from environment or file name
+    :param name: ether file name containing token or environment name which hold the token value.
+    If not provided it will be assumed to read token from IAM_TOKEN environment.
+    :return: token or None
+    """
+    if name and os.path.exists(name):
+        token = None
+        with open(name) as istream:
+            token = istream.read()
+        return token
+    if name:
+        return os.environ.get(name)
+    return os.environ.get("IAM_TOKEN")
+
+
+def tokenData(token, url=None, audUrl=None):
+    """
+    inspect and extract token data
+    :param token: token string
+    :param url: IAM provider URL, by default it is https://cms-auth.web.cern.ch/jwk
+    :param audUrl: audience string, by default it is https://wlcg.cern.ch/jwt/v1/any
+    """
+    if not token or not jwt:
+        return {}
+    if isinstance(token, str):
+        token = bytes(token, 'utf-8')
+    if not url:  # use CMS IAM
+        url = "https://cms-auth.web.cern.ch/jwk"
+    jwksClient = jwt.PyJWKClient(url)
+    signingKey = jwksClient.get_signing_key_from_jwt(token)
+    key = signingKey.key
+    headers = jwt.get_unverified_header(token)
+    alg = headers.get('alg', 'RS256')
+    if not audUrl:  # use WLCG
+        audUrl = "https://wlcg.cern.ch/jwt/v1/any"
+    data = jwt.decode(
+        token,
+        key,
+        algorithms=[alg],
+        audience=audUrl,
+        options={"verify_exp": True},
+        )
+    return data
+
+
+def isValidToken(token):
+    """
+    check if given token is valid or not
+
+    :param token: token string
+    :return: true or false
+    """
+    tokenDict = {}
+    try:
+        tokenDict = tokenData(token)
+    except:
+        traceback.print_exc()
+        raise
+    exp = tokenDict.get('exp', 0)
```

Comment:
Can you please add a comment on what this `exp` attribute is supposed to be?
I guess it stands for `expires`, with an integer EPOCH time of when it expires?

Reply:
done